### ISSUE TITLE
Java-generator fails on empty string

### ISSUE THREAD

**Issue by lmyslinski** [fabric8io/kubernetes-client#4762]
### Describe the bug

I have a CRD which I've obtained from my k8s cluster by running 

`k get crd inferenceservices.serving.kserve.io -o yaml > file.yaml`

The CRD is quite massive so I'll skip including it here. The java generator fails on parsing an enum that's inside the file:

```
    activeModelState:
      default: Pending
      enum:
      - ""
      - Pending
      - Standby
      - Loading
      - Loaded
      - FailedToLoad
      type: string
```

```
Exception in thread "main" java.lang.AssertionError: A string was unexpectedly empty.
	at com.github.javaparser.utils.Utils.assertNonEmpty(Utils.java:69)
	at com.github.javaparser.ast.nodeTypes.NodeWithSimpleName.setName(NodeWithSimpleName.java:42)
	at io.fabric8.java.generator.nodes.JEnum.generateJava(JEnum.java:95)
	at io.fabric8.java.generator.nodes.JObject.generateJava(JObject.java:168)
	at io.fabric8.java.generator.nodes.JObject.generateJava(JObject.java:168)
	at io.fabric8.java.generator.nodes.JObject.generateJava(JObject.java:168)
```

I've forked the client and can patch this, but I want to confirm that it is desired behavior. On one hand, it does make sense to fail on an empty string, but it's a yaml that's generated by k8s so it should probably just skip that value and not throw an error on it.

Please let me know if you'd like to support this behavior. 

### Fabric8 Kubernetes Client version

SNAPSHOT

### Steps to reproduce

No need

### Expected behavior

The CRD is parsed successfully

### Runtime

minikube

### Kubernetes API Server version

1.24

### Environment

macOS

### Additional context

Minikube 1.24.9

**Comment by andreaTP** [comment]
Hi @lmyslinski , thanks a lot for this bug report!

Although this is an edge case I tested that empty strings are valid enum values in CRD and CRs for kubernetes, so we should be able to support this encoding and not skip it (nor fail obviously:-) ).

The bug is easily reproducible by adding an empty string in the `enum-ser-deser` integration test.

I started to try a fix, but got trapped into edge cases and configurations of the Jackson ObjectMapper, which is usually considering empty string as nulls.

Will pick this up again probably next week, meanwhile cc. @shawkins to continue the discussion about the default settings of the  mapper.
